src/libmongoc/src/mongoc/mongoc-topology.c

```diff
@@ -49,12 +49,8 @@ _mongoc_topology_reconcile_add_nodes (mongoc_server_description_t *sd,
    node = mongoc_topology_scanner_get_node (scanner, sd->id);
    if (node) {
       node->hello_ok = sd->hello_ok;
-
-      return true;
-   }
-
-   /* Check if a node for this host was retired in this scan. */
-   if (!mongoc_topology_scanner_has_node_for_host (scanner, &sd->host)) {
+   } else if (!mongoc_topology_scanner_has_node_for_host (scanner, &sd->host)) {
+      /* A node for this host was retired in this scan. */
```

Comment:
This makes the check consistent with what we had before instead of using an early return point.